### ISSUE TITLE
Track duration is sometimes incorrect

### ISSUE THREAD

**Issue by Plaque-fcc** [SimpleMobileTools/Simple-Music-Player#374]
App version: 5.9.0, from Google Play store.
Was the bug introduced with this version: no, observed earlier.

How it acts for me:
I have a playlist (an audio book) with long segments (30 to 120 minutes) and I start it, set the sleep timer for 130 minutes and sometimes fall asleep and the music stops when scheduled or I pause it — doesn't matter.
I use other apps on the device for a while, then I return to the player, it shows the first track in the playlist as Next track in the bottom, which is fine, so I tap it and in the playlist I select a track (its duration is 01:23:15, but the app displays it's 01:23 long, and I actually can even play it from the start or even skip to any part of it.
But it soon stops playing and switches to the next one (and I guess it happens because the app thinks it has ended after these 83 seconds of playback).
If I quit the app (force stop or swipe the window from the open windows list off) and launch it again, it shows proper duration of that track.

**Comment by Plaque-fcc** [comment]
To clarify a bit: this leads to situations when you can have like 37:32 / 01:19 instead of 37:32 / 01:19:22.
In fact, if you just play the track and don't skip or rewind, it will play more than 01:19 of the track.
If you skip before it plays the track for 01:19, it can also play it on.
But if you try to seek to a different piece of the same track after it has been playing for its wrong «max» time displayed, the app won't play it any more and immediately switch to the next track.

**Comment by Plaque-fcc** [comment]
I think I must stress that the duration is being displayed incorrectly only in the track view (as far as I understand, the app uses different view models for current track and track lists), while in the list its duration is being displayed right. Hope it helps to localize the source of the bug.

**Comment by tibbi** [comment]
can you please send a screenshot of what you mean? To know how is the wrong time displayed.

**Comment by Plaque-fcc** [comment]
> can you please send a screenshot of what you mean? To know how is the wrong time displayed.

To be honest, I don't actually care about it any more: the standard player in Android OS 12 finally works just fine for me, so I uninstalled the app.
I appreciate the efforts you put into maintaining simple tools that just work, and I tried looking at the code in attempts to spot (and possibly fix) the source of the problem, but I gave up soon.

At the moment, it works, and for short tracks shouldn't be a problem at all, and shouldn't affect most of your audience. Maybe the steps I provided weren't clear enough for you to reproduce the situation. It is possible that you tried and the bug was already fixed. I don't know.

But if this bug will be encountered by anyone else, it happened for me only on files of more than sixty minutes (MP3). I'm not sure if it's a bug in multimedia collection or not.

At the moment, I have no access to the environment where I discovered this bug, my device received an OTA OS upgrade some time ago.
